.github/workflows/e2e_lighthouse_performance_acceptance_tests.yml

```diff
@@ -125,64 +75,29 @@ jobs:
             requirements_dev.in
             requirements_dev.txt
       - uses: ./.github/actions/merge
-      - name: Cache node modules
-        uses: actions/cache@v3
-        env:
-          cache-name: cache-node-modules
-        with:
-          path: /home/runner/work/oppia/yarn_cache
-          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('yarn.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-build-${{ env.cache-name }}-
-            ${{ runner.os }}-build-
-            ${{ runner.os }}-
-      - name: Attempt to download build files artifact
-        id: download_artifact_1
-        uses: actions/download-artifact@v3
-        continue-on-error: true
-        with:
-          name: build_files
-          path: /home/runner/work/oppia/
-      - name: Unzip build files
-        if: steps.download_artifact_1.outcome != 'failure'
+      # Installing python dependencies becuase we are executing a python script for installing chrome in CI.
+      - name: Install Python deps
         run: |
-          ls -la
-          unzip build_files.zip
-          rm build_files.zip
-          echo "Files in ./:"
-          ls -la .
-          echo "Files in oppia_tools:"
-          ls -la oppia_tools
-          echo "Files in oppia:"
-          ls -la oppia
-          echo "Files in build:"
-          ls -la oppia/build
-          echo "Files in third_party:"
-          ls -la oppia/third_party
-        working-directory: /home/runner/work/oppia
-        shell: bash
-      - uses: ./.github/actions/install-oppia-dependencies
+          pip install --require-hashes --no-deps -r requirements.txt
+          pip install --require-hashes --no-deps -r requirements_dev.txt
       - name: Install chrome
         run: python -m scripts.install_chrome_for_ci
       - name: Install ffmpeg
         run: sudo apt install ffmpeg
-      - name: (Artifact re-build) Cache third_party/static
-        if: steps.download_artifact_1.outcome == 'failure'
-        uses: actions/cache@v3
-        env:
-          cache-name: cache-node-modules
-        with:
-          path: /home/runner/work/oppia/oppia/third_party/static
-          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('dependencies.json') }}
-          restore-keys: |
-            ${{ runner.os }}-build-${{ env.cache-name }}-
-            ${{ runner.os }}-build-
-            ${{ runner.os }}-
-      - name: (Artifact re-build) Build Webpack
-        if: steps.download_artifact_1.outcome == 'failure'
-        run: python -m scripts.build --prod_env
+      # Deleting ubuntu files for getting more space (refer: https://github.com/actions/runner-images/issues/2840#issuecomment-1284059930)
+      - name: Deleting ubuntu files
+        run: |
+          sudo rm -rf /usr/share/dotnet
+          sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+      - name: Initializing Containers
+        run: |
+          sudo make build
+          sudo docker compose up angular-build -d
```

Comment:
I am starting angular-build container independently at first so that it can create the node_modules volume since it is a fresh build. the reason for doing this because we faced an issue during M1 that for the fresh build we cannot start all the containers all at the same time otherwise it will stuck for infinitely. workaround for this is to firstly start the angular-build container to create the node_modules volume which takes the most space, and afterwards start all the other containers

Reply:
>  the reason for doing this because we faced an issue during M1 that for the fresh build we cannot start all the containers all at the same time otherwise it will stuck for infinitely.

DO we know why this was happening and whether it's still happening (after all the optimization we have made after M1)?

Reply:
it was just stuck on the volume creation step. I dont have the logs since this was changed 2-3 months back